internal/net/http/transport/roundtrip.go

```diff
@@ -41,20 +42,37 @@ func NewExpBackoff(opts ...Option) http.RoundTripper {
 	return e
 }
 
+// RoundTrip round trip the HTTP request and return the response.
+// If backoff is not set, the default roundTrip implementation will be used.
+// It round trip the request and returns the response, and return any error occurred.
+// It returns errors.ErrTransportRetryable to indicate if the request is consider as retryable.
 func (e *ert) RoundTrip(req *http.Request) (res *http.Response, err error) {
 	if e.bo == nil {
 		return e.roundTrip(req)
 	}
-	var r interface{}
-	r, err = e.bo.Do(req.Context(), func() (interface{}, error) {
-		return e.roundTrip(req)
+
+	var ferr error
+	r, err := e.bo.Do(req.Context(), func() (res interface{}, err error) {
+		res, err = e.roundTrip(req)
+		if err != nil {
+			// if the error is retryable, return the error and let backoff to retry.
+			if errors.Is(err, errors.ErrTransportRetryable) {
+				return nil, err
+			}
+			// if the error is not retryable, return nil error to terminate the backoff execution
+			ferr = err
+			return nil, nil
+		}
+		return res, nil
 	})
 	if err != nil {
 		return nil, err
 	}
+	if ferr != nil {
+		return nil, ferr
+	}
 
-	var ok bool
-	res, ok = r.(*http.Response)
+	res, ok := r.(*http.Response)
```

Comment:
could you please use named return value?

Reply:
I think it use the named return value. it think it will just create a new instance of `ok bool` but do no create `res`

Reply:
https://github.com/vdaas/vald/pull/589/#discussion_r462709330

Reply:
so it will create a new variable?

Reply:
https://play.golang.org/p/M9-KGxWKdt-